ctest.c

```diff
@@ -149,3 +149,7 @@ ARCH_ARM
 ARCH_ARM64
 #endif
 
+#if defined(__riscv)
```

Comment:
Shouldn't you check for `__riscv_xlen` (32, 64, 128) or better for pointer size?